src/pages/signin/PasswordForm.js

```diff
@@ -102,25 +135,31 @@ class PasswordForm extends React.Component {
                         <Text style={[styles.formLabel]}>{this.props.translate('passwordForm.twoFactorCode')}</Text>
                         <TextInput
                             style={[styles.textInput]}
-                            value={this.state.twoFactorAuthCode}
+                            ref={this.twoFactorAuthCode}
+                            defaultValue={this.twoFactorAuthCode}
                             placeholder={this.props.translate('passwordForm.requiredWhen2FAEnabled')}
                             placeholderTextColor={themeColors.placeholderText}
-                            onChangeText={text => this.setState({twoFactorAuthCode: text})}
+                            onChangeText={this.updateTwoFactorAuthCode}
                             onSubmitEditing={this.validateAndSubmitForm}
                             keyboardType={CONST.KEYBOARD_TYPE.NUMERIC}
                         />
                     </View>
                 )}
 
-                {this.props.account && !_.isEmpty(this.props.account.error) && (
+                {this.state.formError && (
                     <Text style={[styles.formError]}>
-                        {this.props.account.error}
+                        {this.state.formError}
                     </Text>
                 )}
 
-                {this.state.formError && (
+                {!_.isEmpty(this.props.account.error) && (
```

Comment:
Is there a reason you changed this from `{this.props.account && !_.isEmpty(this.props.account.error) && (`

Reply:
I tried to make it consistent with the `LoginForm.js`.
```js
                {this.state.formError && (
                    <Text style={[styles.formError]}>
                        {this.state.formError}
                    </Text>
                )}

                {!_.isEmpty(this.props.account.error) && (
                    <Text style={[styles.formError]}>
                        {this.props.account.error}
                    </Text>
                )}
                {!_.isEmpty(this.props.account.success) && (
                    <Text style={[styles.formSuccess]}>
                        {this.props.account.success}
                    </Text>
                )}

```
And now we're already passing default props on both forms
So, the first `condition` should always be `true`
```js
const defaultProps = {
    account: {
        login: '',
    },
};
```

```js
const defaultProps = {
    account: {
        password: '',
        twoFactorAuthCode: '',
    },
};
```

Reply:
Makes sense! Thanks for clarifying